lib/mocha/parameter_matchers/responds_with.rb

```diff
@@ -22,8 +28,29 @@ module ParameterMatchers
     #   object.expects(:method_1).with(responds_with(:upcase, "BAR"))
     #   object.method_1("foo")
     #   # error raised, because "foo".upcase != "BAR"
-    def responds_with(message, result)
-      RespondsWith.new(message, result)
+    #
+    # @example Actual parameter responds with "FOO" when :upcase is invoked and "oof" when :reverse is invoked.
+    #   object = mock()
+    #   object.expects(:method_1).with(responds_with(:upcase => "FOO", :reverse => "oof"))
+    #   object.method_1("foo")
+    #   # no error raised, because "foo".upcase == "FOO" and "foo".reverse == "oof"
+    def responds_with(*options) # rubocop:disable Metrics/CyclomaticComplexity
+      case options.length
+      when 0
+        raise ArgumentError, 'No arguments. Expecting at least one.'
+      when 1
+        option = options.first
+        raise ArgumentError, 'Argument is not a Hash.' unless option.is_a?(Hash)
+        raise ArgumentError, 'Argument has no entries.' if option.empty?
+
+        messages_vs_results = option
+      when 2
+        messages_vs_results = Hash[*options]
+      else
```

Comment:
mayhaps putting the return value logic into the `when` branch itself would make it a bit more clear? something like:

```ruby
when 1
  # ... argument sanity checks ... #
  matchers = option.map { |message, result| RespondsWith.new(message, result) }
  AllOf.new(*matchers)
when 2
  message, result = options
  RespondsWith.new(message, result)
```

either way thanks for adding this 🙌

Reply:
Ah, yes - that's nicer - thanks for the suggestion! 👍

Reply:
Addressed in https://github.com/freerange/mocha/pull/579/commits/4ce03e9ce44e669863832b8417cf6fe58b0f5ceb